src/ripple/app/tx/impl/XChainBridge.cpp

```diff
@@ -0,0 +1,1627 @@
+            continue;
+        atts.push_back(*att);
+    }
+    if (atts.empty())
+    {
+        return tecXCHAIN_PROOF_UNKNOWN_KEY;
+    }
+
+    XChainCreateAccountAttestations curAtts = [&] {
+        if (sleCID)
+            return XChainCreateAccountAttestations{
+                sleCID->getFieldArray(sfXChainCreateAccountAttestations)};
+        return XChainCreateAccountAttestations{};
+    }();
+
+    auto const rewardAccounts = curAtts.onNewAttestations(
+        &atts[0], &atts[0] + atts.size(), quorum, signersList);
+
+    if (!createCID)
+    {
+        // Modify the object before it's potentially deleted, so the meta data
+        // will include the new attestations
+        if (!sleCID)
+            return tecINTERNAL;
+        sleCID->setFieldArray(
+            sfXChainCreateAccountAttestations, curAtts.toSTArray());
+        psb.update(sleCID);
+    }
+
+    // Account create transactions must happen in order
+    if (rewardAccounts && claimCount + 1 == attBegin->createCount)
+    {
+        auto const r = finalizeClaimHelper(
+            psb,
+            bridgeSpec,
+            attBegin->toCreate,
+            /*dstTag*/ std::nullopt,
+            doorAccount,
+            attBegin->sendingAmount,
+            /*rewardPoolSrc*/ doorAccount,
+            attBegin->rewardAmount,
+            *rewardAccounts,
+            srcChain,
+            sleCID,
+            OnTransferFail::removeClaim,
+            ctx_.journal);
+        if (!isTesSuccess(r))
+        {
+            if (r == tecINTERNAL || r == tecINSUFFICIENT_FUNDS ||
+                isTefFailure(r))
+                return r;
+        }
+        // Move past this claim id even if it fails, so it doesn't block
+        // subsequent claim ids
+        (*sleB)[sfXChainAccountClaimCount] = attBegin->createCount;
+        psb.update(sleB);
+    }
+    else if (createCID)
+    {
+        if (sleCID)
+            return tecINTERNAL;
+
+        auto const sleCID = std::make_shared<SLE>(claimKeylet);
+        (*sleCID)[sfAccount] = doorAccount;
+        (*sleCID)[sfXChainBridge] = bridgeSpec;
+        (*sleCID)[sfXChainAccountCreateCount] = attBegin->createCount;
+        sleCID->setFieldArray(
+            sfXChainCreateAccountAttestations, curAtts.toSTArray());
+
+        // Add to owner directory of the door account
+        auto const page = ctx_.view().dirInsert(
+            keylet::ownerDir(doorAccount),
+            claimKeylet,
+            describeOwnerDir(doorAccount));
+        if (!page)
+            return tecDIR_FULL;
+        (*sleCID)[sfOwnerNode] = *page;
+
+        // Reserve was already checked
+        adjustOwnerCount(psb, sleDoor, 1, ctx_.journal);
+        psb.insert(sleCID);
+        psb.update(sleDoor);
+    }
+
+    psb.apply(ctx_.rawView());
+
+    return tesSUCCESS;
+}
+
+TER
+XChainAddAttestation::doApply()
+{
+    STXChainAttestationBatch const batch = ctx_.tx[sfXChainAttestationBatch];
+
+    std::vector<TER> applyRestuls;
```

Comment:
Is this a typo? Is this supposed to be `applyResults`?

Reply:
Fixed in 91de824b2 [FOLD] Address PR feedback 1: